### PR TITLE
Add source information to disassembly reports

### PR DESCRIPTION
**Pull request by rauls5382**
Disassembly reports generated by pprof -disasm will now include line number
information as generated by objdump. This will make the generated assembly more
readable.

As part of this I've introduced a new assemblyInstruction struct. Previously
the code was reusing the graph.Node to represent assembly instructions but it
seems better to have a dedicated type for this.

### CODE REVIEW

File: internal/report/source.go [hunk 1]
```diff
@@ -204,7 +204,7 @@ func printWebSource(w io.Writer, rpt *Report, obj plugin.ObjTool) error {
 
 // sourceCoordinates returns the lowest and highest line numbers from
 // a set of assembly statements.
-func sourceCoordinates(asm map[int]graph.Nodes) (start, end int) {
+func sourceCoordinates(asm map[int][]assemblyInstruction) (start, end int) {
```
Comment (aalexand) [inline internal/report/source.go]: Nit (not quite in the scope of the PR): sourceLineRange/Span might be a better name.

File: internal/report/report.go [hunk 2]
```diff
@@ -417,10 +446,34 @@ func nodesPerSymbol(ns graph.Nodes, symbols []*objSymbol) map[*objSymbol]graph.N
...
+	return a.cum
+}
+
 // annotateAssembly annotates a set of assembly instructions with a
 // set of samples. It returns a set of nodes to display. base is an
 // offset to adjust the sample addresses.
-func annotateAssembly(insns []plugin.Inst, samples graph.Nodes, base uint64) graph.Nodes {
+func annotateAssembly(insns []plugin.Inst, samples graph.Nodes, base uint64) []assemblyInstruction {
 	// Add end marker to simplify printing loop.
 	insns = append(insns, plugin.Inst{
```
Comment (aalexand) [inline internal/report/report.go]: insts since the type is Inst?

Comment (aalexand) [inline internal/report/report.go]: insts since the type is Inst?

Comment (rauls5382) [inline internal/report/report.go]: Done here and in several other places.

File: internal/report/source.go (hunk 1 above)
Comment (rauls5382) [inline internal/report/source.go]: This is really an assembly instruction, so I prefer the current name.

**Pull request merged**